### ISSUE TITLE
KeyError: 'musicShelfRenderer' in get_history() when watch history disbabled

### ISSUE THREAD

**Issue by mrjones-plip** [sigma67/ytmusicapi#245]
I'm on master as of 909d4d367a4c995a96a56e1d82d7d9a6a4471430 and with this script:

```python
import json
from ytmusicapi import YTMusic
ytm=YTMusic("headers_auth.json")
print(json.dumps(ytm.get_history(), indent=2))
```

I get this error:

```
Traceback (most recent call last):
  File "gethistory.py", line 3, in <module>
    ytm.get_history()
  File "/home/mrjones-plip/Documents/ytmusic-test/venv/lib/python3.8/site-packages/ytmusicapi/mixins/library.py", line 202, in get_history
    data = content['musicShelfRenderer']['contents']
KeyError: 'musicShelfRenderer'
```

Turns out a long time ago I turned off my YTM history.  Clearly I need to turn it on to use the `get_history()` feature!  However, we could at a minimum be defensive [here](https://github.com/sigma67/ytmusicapi/blob/909d4d367a4c995a96a56e1d82d7d9a6a4471430/ytmusicapi/mixins/library.py#L202) with an addition of an `if`:

```python
        songs = []
        for content in results:
            if 'musicShelfRenderer' in content:
                data = content['musicShelfRenderer']['contents']
```

The reason I didn't open a PR with this one line change is maybe we want to be even more helpful and check for the object you get back when history is off?  I have history in there from years ago, which would be confusing to see as "latest", but if there was a warning, that'd be nice!

For completeness, here's the `musicNotifierShelfRenderer` object you get back which is the first in the set of results, thus why it errors first (all IDs truncated in case they have PII):

```json
{
    "musicNotifierShelfRenderer": {
        "title": {
            "runs": [
                {
                    "text": "Turn on your YouTube watch history to keep personalized music up to date."
                }
            ]
        },
        "trackingParams": "CK-SOME-ID-HERE-gYj",
        "isVisible": true,
        "buttons": [
            {
                "buttonRenderer": {
                    "style": "STYLE_DEFAULT",
                    "isDisabled": false,
                    "text": {
                        "runs": [
                            {
                                "text": "Learn more"
                            }
                        ]
                    },
                    "trackingParams": "CKM-SOME-ID-HERE-M=",
                    "command": {
                        "clickTrackingParams": "CKM-SOME-ID-HERE-M=",
                        "urlEndpoint": {
                            "url": "https://support.google.com/youtubemusic/answer/6364666",
                            "target": "TARGET_NEW_WINDOW"
                        }
                    }
                }
            },
            {
                "buttonRenderer": {
                    "style": "STYLE_DEFAULT",
                    "isDisabled": false,
                    "text": {
                        "runs": [
                            {
                                "text": "Turn on history"
                            }
                        ]
                    },
                    "navigationEndpoint": {
                        "clickTrackingParams": "CK-SOME-ID-HERE-iM=",
                        "accountPrivacySettingsEndpoint": {}
                    },
                    "trackingParams": "CK-SOME-ID-HERE-iM="
                }
            }
        ]
    }
}
```

**Comment by sigma67** [comment]
Good suggestion, we should cover this case with an exception.

I pushed some suggested changes [here](https://github.com/sigma67/ytmusicapi/tree/no_watch_history), please verify if they work for you

**Comment by sigma67** [comment]
Closed in 3043116

**Comment by mrjones-plip** [comment]
Awesome sauce! Testing the sad path looks good on master now:

```
Traceback (most recent call last):
  File "gethistory.py", line 4, in <module>
    print(json.dumps(ytm.get_history(), indent=2))
  File "/home/mrjones/Documents/ytmusic/venv/lib/python3.8/site-packages/ytmusicapi/mixins/library.py", line 205, in get_history
    raise Exception(error)
Exception: {'text': 'Turn on your YouTube watch history to keep personalized music up to date.'}
```

Thanks for the quick fix!
